typeql-src/modules/ROOT/pages/fundamentals.adoc

```diff
@@ -396,36 +400,100 @@ For example, given the schema:
 define
 
 group-membership sub relation,
-  relates user-group,
-  relates group-member;
+    relates user-group,
+    relates group-member;
 
 user-group sub entity
-  plays group-membership:user-group;
+    plays group-membership:user-group;
 
 subject sub entity,
-  plays group-membership:group-member;
+    plays group-membership:group-member;
 user sub subject;
 ----
 
 `group-membership` is a `relation` type that defines `user-group` and `group-member` roles.
 The `user-group` role is to be played by a `user-group` entity type whereas the `group-member` role is to be played by
 a `subject` type and all its subtypes entities.
 ====
+// end::relation_type_examples[]
+
+==== Roles
+
+Roles are special internal types used by relations. We can't create an instance of a role in a database.
+But we can set an instance of another type (role player) to play a role in a particular instance of a relation type.
 
 Roles allow a schema to enforce logical constraints on types of role players.
 
-.See the example of a role type constraint
+.See the example of a role player's type constraint
 [%collapsible]
 ====
-For example, a `group-membership` relation cannot associate a `user` type entity with a `file` type entity, because
-`file` type entity can't play any role in a `group-membership` relation.
+For example, given the schema below, a `file` type entity can't play any role in a `group-membership` relation,
+and a `user` type entity can play the `group-member` role of a `group-membership` relation because it inherits it
+from the `subject` type.
+
+[,typeql]
+----
+define
+
+group-membership sub relation,
+    relates user-group,
+    relates group-member;
+
+user-group sub entity
+    plays group-membership:user-group;
+
+subject sub entity,
+    plays group-membership:group-member;
+user sub subject;
+
+file sub entity;
+----
 ====
 
-// end::relation_type_examples[]
+==== Abstract roles
 
-[NOTE]
+[IMPORTANT]
+====
+Since TypeDB v.2.22.0, all roles are concrete, even in <<_abstract_types,abstract>> relations.
 ====
-Roles can be used in queries as a part of a relation even though roles do not have a direct type definition.
+
+.See the example of a concrete role in abstract type
```

Comment:
This is a weak reference, since it's hard to maintain its integrity? You can just leave it at the above note IMO

Reply:
If it's the title can we just say "The example below illustrates concrete roles in abstract relation types"?

Reply:
It's a collapsible title. By default, it is collapsed. So, we have to use an invitation to lure users to click on it. 
I'm trying to use it consistently, starting with "See". For example, "See the example..." or "See examples".
We can discuss how to address these collapsible examples as a part of a major style guide discussion if you want.
Here is how it looks on the website: https://typedb.com/docs/typeql/2.x/fundamentals#_relation_types
<img width="454" alt="Screenshot 2023-09-06 at 17 09 19" src="https://github.com/vaticle/typedb-docs/assets/48120135/459791c4-0ed4-47d4-888c-0ff5ef0e2b36">
<img width="687" alt="Screenshot 2023-09-06 at 17 09 51" src="https://github.com/vaticle/typedb-docs/assets/48120135/ca5e93ae-323c-4a53-ba0b-9612143c363f">

Reply:
Ok that's pretty nice!